### DISCUSSION TITLE
Coming up

### DISCUSSION THREAD

**Discussion by ultrafunkamsterdam** [ultrafunkamsterdam/undetected-chromedriver#1420] [Announcements]
* 11 browser screens.
* No video stretching. What you see is actual speed.
* Bypasses heavily protected betting site and CloudFlare.

* Coming very soon

https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/21027969/980214eb-2b46-4888-8117-8d73280ef942

**Comment by anh300320** [comment]
Amazing! Thank you very much for your effort.

**Comment by himajesty** [comment]
Amazing

**Comment by anoduck** [comment]
@ultrafunkamsterdam is giving us back the internet. Cheers!

**Comment by leesanghyone** [comment]
It's amazing. And I think you're amazing. And thank you!

**Comment by solanav** [comment]
Amazing work

**Comment by luluhoc** [comment]
Amazing

**Comment by NCLnclNCL** [comment]
Great

**Comment by NCLnclNCL** [comment]
No driver oh, it as puppeteer

**Reply by luluhoc** [reply]
What do you mean?

**Reply by NCLnclNCL** [reply]
It use CDP aka chrome devtool protol

**Comment by BRKind** [comment]
you are a legend. amazing work. how does it go so fast. my computer completly lags with 4 maybe 5? is it driver modification to make it run lighter? if so, im done. bro you are insane.

**Comment by damodharheadrun** [comment]
amazing work.

**Comment by lzl15953575** [comment]
When will the version be released? I am very excited...

**Comment by ParhamSobhan** [comment]
best developer i know, keep it upppppppppppppppppppp

**Comment by OliverTrust** [comment]
Awesome

**Comment by yanxiang-wang** [comment]
Great!

**Comment by thebetauser** [comment]
![image](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/56662889/c8e2d48d-edbf-4b81-a187-2d7088d8081d)

**Comment by senzacionale** [comment]
Amazing, great job

**Comment by PLFR23456** [comment]
rly nice !

**Comment by devblack** [comment]
> * 11 browser screens.
> * No video stretching. What you see is actual speed.
> * Bypasses heavily protected betting site and CloudFlare.
> * Coming very soon
> 
>  2023-07-25.21-57-26-86_6.mp4

Please a detailed changelog about the "cf Bypass" will help in future updates.

**Comment by yeucodonvn** [comment]
Can it pass through Datadome?

**Comment by ParhamSobhan** [comment]
Is there any ETA?

**Reply by ParhamSobhan** [reply]
I think it was very late and not very soon what he said :(

**Comment by Avnsx** [comment]
Leon this would come in clutch now, especially if cloudflare struggles to patch the bypass 🫡

**Comment by ghost** [comment]
Release it dawg 😈

**Comment by jakobmesserschmidt** [comment]
I need this asap 🫡🫡

**Reply by Sunda001** [reply]
give donation then

**Reply by ParhamSobhan** [reply]
he does not have any donation link

**Comment by Sunda001** [comment]
ETA?

**Comment by donggoing** [comment]
Amazing

**Comment by jgruenker** [comment]
Thank you for the update, this sounds like really great news.

**Comment by damodharheadrun** [comment]
ETA for this release?

**Comment by ParhamSobhan** [comment]


**Reply by UnicodeApocalypse** [reply]
Not happy ? Do it yourself

**Reply by BRKind** [reply]
Who tf you think you are, if you are not doing any contributions why open your mouth. Coming soon means it's in the work with no eta. Chill tf out and let our man cook up the secret sauce

**Comment by ovciina** [comment]
Wow! have no words, amazing man! THANK YOU!

**Comment by NCLnclNCL** [comment]
When up

**Comment by kaka4348** [comment]
You should be called Cloudflare's killer :)

**Comment by matthewransley** [comment]
is there an eta on this tool?

**Comment by Redrrx** [comment]
Without driver?  it can bypass but is it reliable in general use?

**Comment by jstoxrocky** [comment]
You are a legend brother

**Comment by jifeilong9** [comment]
这是使用了cdp协议吗

**Comment by dxdate** [comment]
@ultrafunkamsterdam When will the release be released?

**Comment by ParhamSobhan** [comment]
any status update?

**Comment by ilike2burnthing** [comment]
FFS can none of you read!? There have been 4 'when is this ready' comments in the last 2 days, 7 in the last week, they have all been ignored! It will be ready when it's ready, just wait.

**Reply by ghost** [reply]
When is it ready?

**Reply by senzacionale** [reply]
@ViaBackwards, I already have it. He sent everyone a private email. Didn't you get it?

**Reply by dxdate** [reply]
> @ViaBackwards, I already have it. He sent everyone a private email. Didn't you get it?

How to get New version?

**Reply by ilike2burnthing** [reply]
Sarcasm!

**Reply by saeidsedqi** [reply]
Please send me the latest version:
sedghi.saeed@gmail.com
Thanks

**Reply by ilike2burnthing** [reply]
1. no
2. don't post your email address on a public site unless you really want spam

**Reply by ghost** [reply]
> @ViaBackwards, I already have it. He sent everyone a private email. Didn't you get it?

Ah, yes I did, thank you! Its amazing... thank you ultra funk amsterdam

**Reply by OliverTrust** [reply]
Thank you for pre release version! AWESOME

**Reply by yanxiang-wang** [reply]
??? I am confused.

**Reply by horror** [reply]
Thx for new version on email. God bless u

**Reply by jifeilong9** [reply]
I haven't received it. Can you send me a copy

**Reply by ilike2burnthing** [reply]
Sent it to Sarcasm@NotReal.lol

**Reply by leesanghyone** [reply]
Me too, please. lsh920910@gmail.com

**Reply by jifeilong9** [reply]
Hello, the email does not exist, causing the message to be returned

**Reply by littlecutebird** [reply]
send it to lolyouguys@so.funny

**Reply by ilike2burnthing** [reply]
This thread is making me lose faith in humanity...

**Reply by jifeilong9** [reply]
可以不提供帮助，但是不要讽刺需要帮助的人
It's okay not to provide help, but don't satirize those in need

**Reply by devblack** [reply]
People are so innocent. They will not send nothing. Just want your mails to spam.

**Reply by ilike2burnthing** [reply]
If you refuse to read, you deserve to be mocked.

**Comment by jstoxrocky** [comment]
I have an actual question...what is meant by 11 browser screens? Is this like some kind of parrallelization (cant spell sry) improvement?

**Comment by emptysuns** [comment]
BRO！！！！！！
You‘re my GOD！！！！

**Comment by Krylanc3lo** [comment]
Looking forward to testing it. Let us know if we can help

**Comment by edelosoft** [comment]
Thanks for new version, how can we obtain a copy of pre-release version?

**Reply by ilike2burnthing** [reply]
https://github.com/ultrafunkamsterdam/undetected-chromedriver/discussions/1420#discussioncomment-6640207

**Reply by edelosoft** [reply]
My bad, skipped the sarcasm part.

**Comment by ThomasTrovon** [comment]
Anxious! Why did the website I use stop working today...

https://stake.com/

😥😥😢😢😔😔

**Comment by shafaitahir8** [comment]
Amazing

**Comment by donggoing** [comment]
The update seems to be working.

**Comment by ParhamSobhan** [comment]
anyone else have problem with the update? because my code still does not work and get stuck on captcha just like before(but it passes the captcha when i close the script!) :(

**Comment by donggoing** [comment]
The current version seems to work. But it can work on CF, not datadome.

**Reply by Gait0fH3ll** [reply]
Hi, what browser version and undetected_driver version are you on?

I'm on:
Version 115.0.5790.171 (Browser)
Starting ChromeDriver 114.0.5735.90 

And cloudflare will not complete.

**Reply by donggoing** [reply]
the latest.
I just find out something wired that it works for me in some sites protected by CF before(i am not sure if they change) but not for https://nowsecure.nl/

**Reply by Krylanc3lo** [reply]
CF is not bypassed on my side, same version than you

**Reply by donggoing** [reply]
Yep. It should be caused by these sites for some reason, doesn't work again.

**Comment by saeidsedqi** [comment]
This update is funny :)) only the version number has been changed

**Reply by ilike2burnthing** [reply]
Nope - https://github.com/ultrafunkamsterdam/undetected-chromedriver/commit/47234b0b13540b6ac15f6b2b65a277e1241f302e

**Reply by saeidsedqi** [reply]
It doesn't work anyway.

**Reply by ThomasTrovon** [reply]
It didn't work with my site here too
https://stake.com/

:'(

**Comment by devCanario** [comment]
@ultrafunkamsterdam what version of chrome are you testing on?

**Comment by sirjo66** [comment]
From chrome version 115 you need to download JSON file from https://googlechromelabs.github.io/chrome-for-testing/known-good-versions-with-downloads.json , then search the nearest version to Chrome installed, and download chromedriver from the correct link.
The link "https://chromedriver.storage.googleapis.com/LATEST_RELEASE_.........." works only until version 114, as you see at https://chromedriver.chromium.org/downloads

**Reply by Gait0fH3ll** [reply]
Hi, I notice my undetected_chromedriver.exe is using 114.. 

So I've deleted it. Upon downloading the latest version (from what you linked)

How do I then update my chromedriver? Which files do I need to replace etc.

Thanks

**Reply by ThomasTrovon** [reply]
I already tried that. From this site

https://googlechromelabs.github.io/chrome-for-testing/

I downloaded the latest version file, today, 115.0.5790.170, from this link:

https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/115.0.5790.170/win64/chromedriver-win64.zip

And put it in the directory where the undetected driver is:

C:\Users\Thomas\AppData\Roaming\undetected_chromedriver

But there's no point in doing that, because when I run my python application, when it starts, it ignores the file in the folder, and downloads the most current one from their repository (114.0.5735.90). So, it will only work if there is a way to skip the file download, and use what we point out to be used.

**Reply by ThomasTrovon** [reply]
I was able to use driver 115.0.5790.170 by doing this:

- After downloading from the link I provided above, I placed the driver .exe file in the same folder as my python application...

![image](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/10778234/058395b9-b898-4059-93f0-bbd42a691259)

- In the code where the library call is made, I put:
driver = uc.Chrome(driver_executable_path='chromedriver.exe')
You don't need to point directory when it's in the same folder. So it takes just the file name.

![image](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/10778234/80ea7ce1-85d0-4e59-b9d9-587b7302a547)

- Still not working (https://stake.com/):

![image](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/10778234/10083ddc-98fd-483f-82c0-837d960f3ccc)

☹☹☹😥😥😥😢😢😢😔😔😔💔💔💔

**Reply by devblack** [reply]
For that website is necessary await 5 seconds or more. An more avance alternative is handling the redirect and get the html content.

**Reply by ThomasTrovon** [reply]
I waited many minutes. Well over 5 seconds. Simply cannot access it. If there is any way that someone can do it, I can only believe it when I see the method that was used. But I still haven't been able to get it.

**Reply by devblack** [reply]
Then the UC is getting detected by Cloudflare (If Cloudflare detected emulation it gives an infinite redirect).

With my private solver is working well, this means Cloudflare are just blocking open-source projects like this.

![image](https://github.com/ultrafunkamsterdam/undetected-chromedriver/assets/33812115/cd063820-9b50-4b1f-be96-10a1376f20a4)

**Reply by Gait0fH3ll** [reply]
Hi, anyway to privately contact you?

**Reply by devblack** [reply]
You can contact me via.
 Telegram: https://t.me/sexyp0w3r
 Discord: devblack_

**Comment by sirjo66** [comment]
no, it don't work because when undetected-chromedriver start, download the chromedriver correct AND change some bytes inside chromedriver.exe (as you see in patcher.py file)
If you download chromedriver.exe from correct link ok, but you need also modify exe file

**Comment by damodharheadrun** [comment]
I lost my hope, waiting more than 3 weeks for the new version

**Reply by senzacionale** [reply]
Then you do it faster.

**Reply by damodharheadrun** [reply]
Join with me will do it together

**Reply by NCLnclNCL** [reply]
Haha

**Comment by w956205480** [comment]
Is the new version online?

**Comment by ultrafunkamsterdam** [comment]
> I already tried that. From this site
> 
> https://googlechromelabs.github.io/chrome-for-testing/
> 
> I downloaded the latest version file, today, 115.0.5790.170, from this link:
> 
> https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/115.0.5790.170/win64/chromedriver-win64.zip
> 
> And put it in the directory where the undetected driver is:
> 
> C:\Users\Thomas\AppData\Roaming\undetected_chromedriver
> 
> But there's no point in doing that, because when I run my python application, when it starts, it ignores the file in the folder, and downloads the most current one from their repository (114.0.5735.90). So, it will only work if there is a way to skip the file download, and use what we point out to be used.

It simply won't work with beta or canary releases of chrome. As you can see on https://chromedriver.storage.googleapis.com the lastest official Chrome driver release is currently 114. Previously, they also had 1 version up from the current stable, but lately they seen to walk a bit behind.

**Reply by ThomasTrovon** [reply]
@ultrafunkamsterdam 
The new version worked perfectly here! Thank you very much friend. You are a hero! 💪💪💪🤝🤝🤝

**Comment by devCanario** [comment]
@ultrafunkamsterdam any updates? if it helps i would recomend removing the console.log from the patch and just patch it with a string, it looks like cloudflare is listening on the console logs

**Reply by NCLnclNCL** [reply]
How to patch

**Reply by devCanario** [reply]
use hxd editor and search for console.log("undetected chromedriver") and replace it "undetected chromedriver"

**Comment by estromenko** [comment]
Any updates?

**Reply by ultrafunkamsterdam** [reply]
Yes! 
I will add it in a couple of days. Rounding up documentation and refactoring. Still in dubio whether to add it or sep repo.

**Comment by BRKind** [comment]
It's working for me, bypassing cloudfare but headless seems to not work. If anyone finds a solution, send me private email on spomaky@gmail.com
